c7n/resources/awslambda.py

```diff
@@ -142,7 +183,98 @@ def _augment(r):
             resources, event)
 
 
-@AWSLambda.action_registry.register('delete')
+@actions.register('mark-for-op')
+class TagDelayedAction(TagDelayedAction):
+    """Action to specify an action to occur at a later date
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: lambda-delete-unused
+                resource: lambda
+                filters:
+                  - "tag:custodian_cleanup": absent
+                actions:
+                  - type: mark-for-op
+                    tag: custodian_cleanup
+                    msg: "Unused lambda"
+                    op: delete
+                    days: 7
+    """
+
+    permissions = ('lambda:TagResource',)
+
+    def process_resource_set(self, functions, tags):
+        client = local_session(self.manager.session_factory).client('lambda')
+        tag_dict = {}
+        for t in tags:
+            tag_dict[t['Key']] = t['Value']
+        for f in functions:
+            arn = f['FunctionArn']
+            client.tag_resource(Resource=arn, Tags=tag_dict)
```

Comment:
there's a behavior issue i'm noticing on older lambdas where they don't support tags till they have a new code revision. this will need a try except wrapper, we may need to do a separate pull request for an increment revision  on the lambda source to get the updated tagging capability.